### PR TITLE
[DependencyInjection] Deprecate numeric parameter names

### PR DESCRIPTION
**Pull request by HeahDude**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | ~
| License       | MIT
| Doc PR        | TODO

While trying to use `'.' !== $key[0]` instead of `str_starts_with($key, '.')` in #47680, I noticed some tests were failing due to the usage of numeric parameter names in the fixtures.

This leads to inconsistent behavior since the following code: `$parameterBag->set(10, 10)`, will first cast the name `10` to string because of the method signature, but will then cast back to integer when using it as an array key in https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php#L89.
Because Symfony does not use strict types, it can be really tricky.

This PR propose to deprecate using such names to be able to properly throw in 7.0.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php [hunk 1]
```diff
@@ -86,6 +86,10 @@ public function get(string $name): array|bool|string|int|float|\UnitEnum|null
 
     public function set(string $name, array|bool|string|int|float|\UnitEnum|null $value)
     {
+        if (is_numeric($name)) {
+            trigger_deprecation('symfony/dependency-injection', '6.2', 'Using numeric parameter names is deprecated and will throw as of 7.0.');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php]: let's give the name of the offending parameter:
`'Using numeric parameter name "%s" is deprecated and will throw as of 7.0.', $name);`
can you also please add the throwing line commented to make it easy to upgrade on 7.0?

**Review by nicolas-grekas — COMMENTED** [review]
LGMT, just some minor comment

File: src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php (hunk 1 above)
Comment (HeahDude) [reply src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php]: Done.

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ CHANGELOG
  * Allow #[When] to be extended
  * Change the signature of `ContainerAwareInterface::setContainer()` to `setContainer(?ContainerInterface)`
  * Deprecate calling `ContainerAwareTrait::setContainer()` without arguments
+ * Deprecate using numeric parameter names
```
Comment (chalasr) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md:14]: Ah, can you also update `UPGRADE-6.2.md`?

Comment (HeahDude) [reply src/Symfony/Component/DependencyInjection/CHANGELOG.md:14]: Done :).

**Comment by chalasr** [comment]
Thank you @HeahDude.

**Pull request merged**